deploy_docs.sh

```diff
@@ -0,0 +1,86 @@
+#!/bin/bash
+set -e # Exit with nonzero exit code if anything fails
+
+# Build the documentation from the SOURCE_BRANCH
+# and push it to TARGET_BRANCH.
+SOURCE_BRANCH="master"
+TARGET_BRANCH="gh-pages"
+
+# Pull requests and commits to other branches shouldn't try to deploy
+#if [ "$TRAVIS_PULL_REQUEST" != "false" -o "$TRAVIS_BRANCH" != "$SOURCE_BRANCH" ]; then
+#    echo "Skipping deploy on $TRAVIS_BRANCH. We only deploy docs automatically from development."
+#    exit 0
+#fi
+
+# Save some useful information
+REPO=`git config remote.origin.url`
+SSH_REPO=${REPO/https:\/\/github.com\//git@github.com:}
+SHA=`git rev-parse --verify HEAD`
+
+# Add rsa keys to the ssh agent to push to GitHub
+#gpg --output ../id_tardis_rsa --batch --passphrase $DECRYPT_GITHUB_AUTH --decrypt id_tardis_rsa.enc
+#chmod 600 ../id_tardis_rsa
+#if [ -z "$SSH_AUTH_SOCK" ] ; then
+#  eval `ssh-agent -s`
+#  ssh-add ~/.ssh/id_azure_rsa
+#fi
+
+
+# Clone the existing gh-pages for this repo into out/
+# Create a new empty branch if gh-pages doesn't exist yet (should only happen on first deply)
+git clone $REPO out
+cd out
+
+# # Regenerate the API documentation
+# git checkout $SOURCE_BRANCH || git checkout --orphan $SOURCE_BRANCH
+# sphinx-apidoc -f -M -o docs/source/ .
+
+# # Add new docs to the repo
+# git add docs/source --all
+
+# # Commit and push to SOURCE_BRANCH
+# git commit -m "Regenerate API Documentation: ${SHA}" || true
+# git push $SSH_REPO $SOURCE_BRANCH || true
+
+git checkout $TARGET_BRANCH || git checkout --orphan $TARGET_BRANCH
+cd ..
+
+# Clean out existing contents
+rm -rf out/docs/** || exit 0
+
+# Pull from SOURCE_BRANCH again
+git pull || true
+
+# Build the Sphinx documentation
+cd docs
+make html
+cd ../  
+
+mkdir -p out/docs/
+mv -f docs/_build/html/* out/docs/
+touch out/.nojekyll
+
+# Now let's go have some fun with the cloned repo
+cd out
+git config --local user.name "Azure Pipelines"
+git config --local user.email "azuredevops@microsoft.com"
+
+echo "doing git add/commit/push"
+
+# Commit the "changes", i.e. the new version.
+# The delta will show diffs between new and old versions.
+git add --all
```

Comment:
are the files then not in the `out/` directory?

Reply:
The files should be in both the out/ directory and the vm(which won't commit them), it does cd once more into out/ before this command at line 64, or did I misinterpet what you meant?

Reply:
what do you mean by the vm

Reply:
I meant to say the source branch. A.k.a the master that is being run in the vm, which makes the out directory, which github tells me is a repo, that is also checkout out to the gh-pages branch, that is  pushed to github independent of the original source branch. 
Github gives me this message also:
warning: adding embedded git repository: out
hint: You've added another git repository inside your current repository.